### PR TITLE
Add publising to tf registry

### PR DESCRIPTION
**Pull request by graemedavidson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,10 +47,11 @@ provider "k8s" {
 ```
 
 **WARNING:** Configuration from the variable will be recorded into a temporary file and the file will be removed as
-soon as call is completed. This may impact performance if the code runs on a shared system because
-and the global tempdir is used.
+soon as call is completed. This may impact performance if the code runs on a shared system because and the global
```
Comment (sedan07) [inline README.md]: "because and the"
remove "and" maybe? or is there a bit missing?

File: README.md [hunk 2]
```diff
@@ -37,10 +47,11 @@ provider "k8s" {
...
+tempdir is used.
 
-The k8s Terraform provider introduces a single Terraform resource, a `k8s_manifest`. The resource contains a `content` field, which contains a raw manifest.
+The k8s Terraform provider introduces a single Terraform resource, a `k8s_manifest`. The resource contains a `content`
```
Comment (sedan07) [inline README.md]: doesn't it also need a `name` and `namespace` (if a NS'd resource)?

File: README.md [hunk 3]
```diff
@@ -1,23 +1,41 @@
 # Kubernetes Terraform Provider
 
-The k8s Terraform provider enables Terraform to deploy Kubernetes resources. Unlike the [official Kubernetes provider][kubernetes-provider] it handles raw manifests, leveraging `kubectl` directly to allow developers to work with any Kubernetes resource natively.
+The k8s Terraform provider enables Terraform to deploy Kubernetes resources. Unlike the
+[official Kubernetes provider][kubernetes-provider] it handles raw manifests, leveraging `kubectl` directly to allow
+developers to work with any Kubernetes resource natively.
 
-## Usage
+This provider is published in the [Terraform Registry Provider](https://registry.terraform.io/providers/fiveai/k8s)
```
Comment (sedan07) [inline README.md]: Probably just call it the "Terrform Registry"

File: README.md [hunk 4]
```diff
@@ -37,48 +55,67 @@ provider "k8s" {
...
 ```
 
-In this case `manifests/nginx-deployment.yaml` is a templated deployment manifest.
+The templated resource definition should then resemble the following example.
 
 ```yaml
-apiVersion: apps/v1beta2
-kind: Deployment
+apiVersion: apps/v1
+kind: ${kind}
```
Comment (sedan07) [inline README.md]: where does `kind` and `app` vars come from?

Comment (graemedavidson) [reply README.md]: Kind is the type of deployment, which I regularly use just to make tf easier to read without going into the template, though then usually it is also in the name. App is just something from the original code and did not want to just remove.

Comment (graemedavidson) [reply README.md]: just realised I have done it wrong, wait a min

File: README.md [hunk 5]
```diff
@@ -37,48 +55,47 @@ provider "k8s" {
...
-  template = "${file("manifests/nginx-deployment.yaml")}"
+The terraform resource definition should include:
 
-  vars {
-    replicas = "${var.replicas}"
-  }
-}
+* `name` of the resource
+* `namespace` to be deployed to
+* `content` resource to be deployed
```
Comment (sedan07) [inline README.md:69]: Doesn't it require `kind` too? and `namespace` is only needed for resources which are namespaced so not cluster resources.

Comment (graemedavidson) [reply README.md:69]: oops, forgot I removed that. Added your description for namespace

**Review by sedan07 — APPROVED** [review]

**Pull request merged**